doc/radosgw/uadk-accel.rst

```diff
@@ -0,0 +1,131 @@
+===============================================
+UADK Acceleration for Compression
+===============================================
+
+UADK is a framework for user application to access hardware accelerator in a
+unified, secure, and efficient way. UADK is comprised of UACCE, libwd and many
+other algorithm libraries for different applications.
+
+See `Compressor UADK Support`_.
+
+
+UADK in the Software Stack
+=========================
+
+UADK is a general-purpose user space accelerator framework that uses shared
+virtual addressing (SVA) to provide a unified programming interface for hardware
+acceleration of cryptographic and compression algorithms.
+
+UADK includes Unified/User-space-access-intended Accelerator Framework (UACCE),
+which enables hardware accelerators from different vendors that support SVA to
+adapt to UADK.
+
+Currently, HiSilicon Kunpeng hardware accelerators have been registered with
+UACCE. Through the UADK framework, users can run cryptographic and compression
+algorithms using hardware accelerators instead of CPUs, freeing up CPU computing
+power and improving computing performance.
+
+A user can access the hardware accelerators by performing user-mode operations on
+the character devices, or the use of UADK can be done via frameworks that have
+been enabled by others including UADK support (for example, OpenSSL* libcrypto*,
+DPDK, and the Linux* Kernel Crypto Framework).
+
+See `OpenSSL UADK Engine`_.
+
+UADK Environment Setup
+=====================
+UADK consists of UACCE, vendors’ drivers, and an algorithm layer. UADK requires the
+hardware accelerator to support SVA, and the operating system to support IOMMU and
```

Comment:
So if one currently disables IOMMU on EPYCs for better performance, one would want to try it both ways to see which is faster?

Reply:
More focus on offloading CPU usage now. Will try later.